packages/commonwealth/client/scripts/controllers/server/notifications.ts

```diff
@@ -82,29 +84,43 @@ class NotificationsController {
   }
 
   private _subscriptions: NotificationSubscription[] = [];
-  public get subscriptions() {
-    return this._subscriptions;
+
+  public get discussionSubscriptions(): NotificationSubscription[] {
+    return this._subscriptions.filter(
+      (s) => s.categoryId !== NotificationCategories.ChainEvent
+    );
   }
 
-  public subscribe(category: string, objectId: string) {
-    const subscription = this.subscriptions.find(
-      (v) => v.category === category && v.objectId === objectId
+  public get chainEventSubscriptions(): NotificationSubscription[] {
+    return this._subscriptions.filter(
+      (s) => s.categoryId === NotificationCategories.ChainEvent
     );
+  }
+
+  public subscribe(data: SubUniqueData) {
+    const subscription = this.findNotificationSubscription(data);
     if (subscription) {
       return this.enableSubscriptions([subscription]);
     } else {
+      const requestData = Object.fromEntries(
+        Object.entries(data.options).map(([k, v]) => [
```

Comment:
What does this regex do?

Reply:
@timolegros please explicitly map the values if possible -- just for clarity's sake. I feel like this sort of automatic clever code can lead to mysterious bugs down the road. #4703 def worth pursuing in the future, or something similar.

Reply:
Done.